shared/app/nav.desktop.js

```diff
@@ -54,6 +54,17 @@ const mapDispatchToProps = (dispatch: Dispatch, ownProps: OwnProps) => ({
       // clicking the chat tab when already selected should do nothing.
       return
     }
+
+    // If we're going to the profile tab, switch to the current user's
+    // profile first before switching tabs, if necessary.
+    if (tab === profileTab) {
+      dispatch(navigateTo([], [profileTab]))
+      if (ownProps.routeSelected !== profileTab) {
```

Comment:
i think you can just do this and it'll be benign if the route is the same right?